coco/request_forwarder.py

```diff
@@ -54,6 +56,40 @@ def add_endpoint(self, name, endpoint):
         """
         self._endpoints[name] = endpoint
 
+    # We need a separate server to track metrics produced by this process.
+    # This should be called by the worker process when it is started.
+    @staticmethod
+    def start_prometheus_server(port=9090):
+        """
+        Start prometheus server.
+
+        Parameters
+        ----------
+        port : int
+            Server port.
+        """
+        start_http_server(port)
+
+    def init_metrics(self):
+        """
+        Initialise success/failure counters for every prometheus endpoint.
+        """
+        self.counter_succ = {}
+        self.counter_fail = {}
+        for edpt in self._endpoints:
+            cnt_succ = Counter(format_metric_name(f"coco_{edpt}_success"),
```

Comment:
I should have thought about that more: The extra label on the failure counter should really be the HTTP code. This will be 200 in the success case and therefor we only need one metric.